source/priority-queue.ts

```diff
@@ -32,6 +33,20 @@ export default class PriorityQueue implements Queue<RunFunction, PriorityQueueOp
 		this.#queue.splice(index, 0, element);
 	}
 
+	setPriority(id: string, priority: number) {
+		const existingIndex: number = this.#queue.findIndex((element: Readonly<PriorityQueueOptions>) => element.id === id);
+		if (existingIndex === -1) {
+			throw new Error('Invalid Index - No promise function of specified id available in the queue.');
+		}
+
+		const [item] = this.#queue.splice(existingIndex, 1);
+		if (item === undefined) {
+			throw new Error('Undefined Item - No promise function of specified id available in the queue.');
+		}
```

Comment:
I don't think these should throw. It should be allowed to set a priority without knowing beforehand whether the has already finished.